File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/movable/MovableObject.java
package com.agrogames.islandsofwar.engine.abs.movable;

import com.agrogames.islandsofwar.engine.abs.common.Cell;
import com.agrogames.islandsofwar.engine.abs.unit.IUnit;

public interface MovableObject extends IUnit {
    void setGoal(Cell goal);
}

File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/movable/MovableObject.kt
package com.agrogames.islandsofwar.engine.abs.movable

import com.agrogames.islandsofwar.engine.abs.common.Cell
import com.agrogames.islandsofwar.engine.abs.unit.IUnit

interface MovableObject : IUnit {

    fun cleverSetGoal(goal: Cell)
    val route: Collection<Cell>
}